### PR TITLE
MGDBOR-363: Refactor Bridge provisioning to re-use resilient mechanism 

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-363](https://issues.redhat.com/browse/MGDOBR-363) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/BridgeWorker.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        // Options:
+        // (1) Make "KafkaTopic" a "first class dependency" (i.e. ManagedResource), DB table etc etc so we can block
+        //     the Shard from creating the Bridge until this has completed successfully. This would also be a change
+        //     to ConnectorWorker to depend on the "KafkaTopicManagedResource". TBH this seems a good idea. We'd
+        //     have a DB table for ALL KafkaTopics we've created and hence be able to clean-up better...
+        //     Which is then a leading question for Luca's idea of a generic "Resource" table vs discrete ones
+        //     for Bridge, Processor, Connector, KafkaTopics....
+        // (2) Close JIRA MGDOBR-363, don't have Bridges provisioned with Workers. Use of RHOAS then lacks
+        //     support of a re-try or timeout mechanism, provisioning of resources (i.e. KafkaTopic) for Bridges
+        //     looses any resilience; blah blah blah.
```
Comment (manstis) [inline manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/BridgeWorker.java:61]: Hey @lucamolteni @robpblake @danielezonca Don't be afraid of this PR.. it's a draft and this comment I want to discuss with you..... My opinion is to make "KafkaTopic" a first class `ManagedResource` so we can have the `Worker` mechanism treat it as a fallible dependency and hence be subject to re-tries etc.

This JIRA is for Sprint 216 but before I go running off thought to ask your opinions.. happy to set up a meeting for a chat if you like :smile: :coffee: :computer:

Comment (lucamolteni) [reply manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/BridgeWorker.java:61]: I'm all for it, eventually I'd like us to provision all kinds of resources in the same way

Comment (robpblake) [reply manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/BridgeWorker.java:61]: @manstis @lucamolteni I'm not following the discussion here. Why is using the worker pattern to provision the topic/service account for a bridge with our current implementation a problem?

In particular this comment:

```
 The Shard will therefore potentially
        // detect that there is a Bridge in ACCEPTED state and start provisioning it in Kubernetes before this call completes.
```
Our query to expose Bridges to provision to the shard will be `from bridge b where b.status="AVAILABLE" and b.dependencyStatus="READY"` ? What am I missing?

Comment (lucamolteni) [reply manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/BridgeWorker.java:61]: I'm not following either @robpblake I think this is a good idea. Where did you find that comment?

Comment (manstis) [reply manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/BridgeWorker.java]: I can't even begin to explain how my mind was working when I wrote that comment :rofl: 

I'd clearly forgotten about `dependencyStatus`.

Anyway, PR updated and ready for your reviews.

**Comment by manstis** [comment]
@robpblake @lucamolteni @jomarko @danielezonca Any takers for reviewing this PR? :smiley:

**Review by robpblake — APPROVED** [review]
LGTM. Thanks @manstis

**Review by danielezonca — APPROVED** [review]

**Pull request merged**